dwelling_draft.R

```diff
@@ -0,0 +1,339 @@
+##Housing indicators
+
+# load 
+library(readr)
+library(janitor)
+library(tidyverse)
+library(readxl)
+library(scales)
+library(openxlsx)
+library(ggplot2)
+library(phsstyles)
+library(tidylog)
+library(stringr)
+library(purrr)
+
+################################################################################-----------------------------------------------
+
+
+# Set file paths for folders
+fp_cpp <- "//conf/LIST_analytics/West Hub/02 - Scaled Up Work/CPP Community Profiles/Data Extracts/"
+
+# All areas lookup
+lookup <- read_rds("//conf/LIST_analytics/West Hub/02 - Scaled Up Work/CPP Community Profiles/Data Extracts/dz_pop_scotpho_scripts/DataZone11_All_Geographies_Lookup.rds")
+  
+
+# Long lookup with all codes of interest
+long_lookup <- lookup %>% 
+  select(datazone2011,intzone2011, ca2019,hscp2019,hb2019,hscp_locality) 
+
+
+# Set rounding function
+rounding <- function(x){
+  case_when(x < 1 ~ round_half_up(x, 2),
+            between(x, 1, 1000) ~ round_half_up(x, 1),
+            x > 1000 ~ round_half_up(x, 0))
+}
+
+# Turn off scientific notation
+options(scipen = 999)
+
+# System unmask function so files have read-write permissions
+Sys.umask("006")
+
+
+
+#################### Section 2: Data imports and cleaning ######################
+
+## Dwelling estimates and occupants---------------------------------------------
+
+# Path to your Excel file
+file_path <- "//conf/LIST_analytics/West Hub/02 - Scaled Up Work/CPP Community Profiles/Data Extracts/dz_pop_scotpho_scripts/household_estimates.xlsx"
+
+## identify sheet names 
+sheets <- excel_sheets(file_path) %>%  str_subset(pattern = "^\\d+$")
+
+# read in data for years available
+dwelling_estimates <-
+  map_df(sheets, ~read_excel(paste0("//conf/LIST_analytics/West Hub/02 - Scaled Up Work/CPP Community Profiles/Data Extracts/dz_pop_scotpho_scripts/household_estimates.xlsx"),
+                                            sheet =.x,
+                                            skip = 3) %>%
+           mutate(year = as.integer(.x))) %>%
+           clean_names()
+
+
+# clean data
+dwelling_estimates <- dwelling_estimates %>% 
+  clean_names() %>% 
+  select(c(year, "area_code" = data_zone_code, council_area_code, "total_dwellings" = total_number_of_dwellings,
+           occupied_dwellings, 
+           occupied_dwellings_exempt_from_paying_council_tax))
+
+
+#Get Scotland numbers
```

Comment:
Instead of aggregating to each geography level separately, could simplify the code by joining on the lookup and using pivot_longer to get the data in long format with all geography levels stacked and then aggregating up